### PR TITLE
Fix nested operations when no delete, connect or disconnect remaining

### LINKED ISSUE #3816: Error in schema generation when using nestedOperations
**Describe the bug**
There is an error in schema generation if you restrict a bidirectional relationship only on one side with nestedOperations.

**Type definitions**
````graphql
type Movie {
  name: String!
  genre: Genre! @relationship(type: "HAS_GENRE", direction: OUT, nestedOperations: [CONNECT, DISCONNECT])
}

type Genre {
  name: String!
  movies: [Movie!]! @relationship(type: "HAS_GENRE", direction: IN)
}
````

**To Reproduce**

1. Run a server with the following code...
2. See error
````
Error: Type with name "MovieDeleteInput" does not exists
````

**System (please complete the following information):**

* Version: @neo4j/graphql@3.24.1

**Additional context**
It works if you restrict nestedOperations to [CONNECT, DISCONNECT] on both sides.

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

There was a bug where we would be referring to non-existent types when generating relationship operations. This fix adds a check that there is at least one relationship from a node with matching nested operations.

## Complexity

Complexity: Low

# Issue

Closes #3816

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3828.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 33 | 52 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 79 | 114 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 51 | 74 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3127 | 3195 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 2854 | 2747 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 55 | 49 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 42 | 59 | 2174 |
|  aggregations.DeeplyNestedCount | 13070335 | 13070331 | 4215 | 4325 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 128 | 173 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 74 | 240 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 289 | 184 | 3003 |
|  connections.Connection | 14082 | 14082 | 111 | 69 | 2174 |
|  connections.NestedConnection | 45466 | 45470 | 115 | 157 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 67 | 87 | 1040 |
|  connections.ConnectionWithSortAndCypher | 3284 | 3284 | 196 | 349 | 1040 |
|  create.SimpleMutation | 7 | 7 | 59 | 103 | 1 |
|  cypher-directive.TopLevelCypherDirective | 122100 | 122100 | 600 | 695 | 12241 |
|  cypher-directive.TopLevelCypherDirectiveWithColumnName | 153619 | 153617 | 184 | 179 | 12241 |
|  delete.SimpleDelete | 19401 | 19401 | 231 | 257 | 1040 |
|  delete.NestedDeleteInUpdate | 23202 | 23202 | 182 | 232 | 2040 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 60 | 57 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 117 | 133 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 40 | 48 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 75 | 85 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 29 | 27 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 46 | 57 | 2174 |
|  query.QueryWhere | 9710 | 9697 | 36 | 45 | 2165 |
|  query.SimpleQueryWithNestedWhere | 9888 | 9875 | 54 | 61 | 2165 |
|  query.Nested | 10096041 | 10096041 | 11390 | 10927 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 11102 | 11145 | 2004000 |
|  query.OrFilterOnRelationships | 41835 | 41838 | 200 | 276 | 1591 |
|  query.OrFilterOnRelationshipsAndNested | 35746 | 36412 | 229 | 302 | 1591 |
|  query.QueryWithNestedIn | 14423 | 13241 | 58 | 67 | 1557 |
|  query.NestedConnectionWhere | 9834 | 9834 | 63 | 88 | 2174 |
|  query.DeeplyNestedConnectionWhere | 10011 | 10044 | 78 | 106 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19224 | 19320 | 136 | 187 | 1624 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 174 | 236 | 1134 |
|  query.Fulltext | 80 | 80 | 34 | 38 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 46 | 61 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 72 | 90 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 1478 | 1430 | 186 | 354 | 1040 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 54 | 68 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 95 | 117 | 4516 |
|  sorting.SortWithTopLevelCypher | 3121 | 3121 | 59 | 108 | 1040 |
|  unions.SimpleUnionQuery | 321 | 321 | 79 | 87 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 83 | 91 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 349 | 348 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 294 | 323 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 100 | 165 | 2002 |

</details>
        
Old Schema Generation: 39.446s
Schema Generation: 39.716s
Old Subgraph Schema Generation: 50.026s
Subgraph Schema Generation: 49.590s

File: packages/graphql/src/schema/create-relationship-fields/create-relationship-fields.ts [hunk 1]
```diff
@@ -463,4 +464,9 @@ function createRelationshipFields({
     });
 }
 
+function nodeHasRelationshipWithNestedOperation(node: Node, nestedOperation: RelationshipNestedOperationsOption): boolean {
+ return Boolean(node.relationFields.filter(relationField => relationField.nestedOperations.includes(nestedOperation)).length)
+}
```
Comment (mjfwebb) [inline packages/graphql/src/schema/create-relationship-fields/create-relationship-fields.ts:469]:
```suggestion
function nodeHasRelationshipWithNestedOperation(
    node: Node,
    nestedOperation: RelationshipNestedOperationsOption
): boolean {
    return node.relationFields.some((relationField) => relationField.nestedOperations.includes(nestedOperation));
}
```

A slight modification of this to use .some instead. That way we exit the loop early and don't need to do a boolean type conversion.

**Review by mjfwebb — COMMENTED** [review]
Nice! Just one small code change suggestion :)

**Pull request merged**